server/zanata-war/src/main/java/org/zanata/adapter/JsonAdapter.java

```diff
@@ -46,4 +48,9 @@ protected void updateParamsWithDefaults(IParameters params) {
         Parameters p = (Parameters) params;
         p.fromString("extractIsolatedStrings.b=true");
     }
+
+    @Override
+    protected String getIdFor(TextUnit tu, String content, String subDocName) {
+        return HashUtil.generateHash(tu.getName().concat(content));
```

Comment:
If you don't use a global key, won't some of your existing test cases (test-json-duplicateids.json) be merging similar strings into one text flow? I thought you wanted to avoid that.

CopyTrans should work if you tell it to ignore context, **in theory**.